engine.go

```diff
@@ -517,11 +553,13 @@ func (e *RTCEngine) resumeConnection() error {
 	e.client.Start()
 
 	// send offer if publisher enabled
+	e.pclock.Lock()
 	if !e.subscriberPrimary || e.hasPublish.Load() {
 		e.publisher.createAndSendOffer(&webrtc.OfferOptions{
```

Comment:
also a bit concerned about locking these operations that may perform IO/locks other resources.